src/annotations/api/index.ts

```diff
@@ -63,10 +65,10 @@ export const getAnnotations = async (
   }
   const res = await getAnnotationsApi(params)
   if (res.status >= 300) {
-    throw new Error(res.data?.message)
+    throw new Error((res.data as ResponseError).message)
   }
 
-  return res.data.map((retrievedAnnotation: AnnotationResponse) => ({
+  return (res.data as AnnotationList).map((retrievedAnnotation: any) => ({
```

Comment:
Not sure if changing the `AnnotationResponse` to `any` was the way to go here. The IDE says that the items in `AnnotationList` are not of type `AnnotationResponse` because `stream` and `annotations` properties are optional. @hoorayimhelping thoughts on this?

Reply:
Also the `as AnnotationList` casting is required because the IDE says the type is actually `AnnotationList | ResponseError`

Reply:
yes, openapi#154 caused issues when I tested them in #1811 I understood someone was going to update the tests for the actual response coming from annotations and we could merge the openapi pr. when 1811 got closed, i went ahead and merged 154 understanding that some changes to the ui tests would need to be made when y'all pulled in the latest spec.

Reply:
@hoorayimhelping we good to move forward with this change? Any changes you'd like to propose?

Reply:
@genehynson I'd prefer we didn't type a response that used to have a type as `any`. I'm still not quite following these changes @glinton - it seems like we've actually lost resolution here instead of gained it.

Reply:
Sounds like that might be related to https://github.com/influxdata/oats/issues/24, where `any` is used on a referenced type. The changes to openapi define the actual [response structure](https://github.com/influxdata/openapi/pull/154/files#diff-8c843eba1d4939da04c779f4c3de77308fa51dc7dc48176fb0be042c4d9a28a9R342-R352). Because the openapi definition [references the type](https://github.com/influxdata/openapi/pull/154/files#diff-8c843eba1d4939da04c779f4c3de77308fa51dc7dc48176fb0be042c4d9a28a9R352) for the array, oats must be marking it as `any`.

Any news on whether oats will get some attention or if y'all will be switching to a better generator?

Reply:
maybe explicitly marking them as `required` in swagger would help, but that seems a bit pedantic. wouldn't be the worst thing, but it feels odd to mark something as "required" where it's a response structure rather than request :thinking:

Reply:
Tried marking them "required" in the swagger on this branch and it didn't change the generated code. https://github.com/influxdata/openapi/tree/genehynson/trying-something-annotations-2

Reply:
Just noticed that as well

Reply:
@hoorayimhelping would it be ok if I merged this in to unblock other changes? I could open an issue to follow up on this specific typing problem

Reply:
Yeah, go for it and merge @genehynson. Don't let this discussion block you